client/memory.cpp

```diff
@@ -84,6 +85,47 @@ clEnqueueFillBuffer(cl_command_queue command_queue, cl_mem buffer,
 	}
 }
 
+SO_EXPORT CL_API_ENTRY void* CL_API_CALL
+clEnqueueMapBuffer(cl_command_queue command_queue, cl_mem buffer, cl_bool blocking_map,
+                   cl_map_flags map_flags, size_t offset, size_t size,
+                   cl_uint num_events_in_wait_list, const cl_event* event_wait_list,
+                   cl_event* event, cl_int *errcode_ret) CL_API_SUFFIX__VERSION_1_0
+{
+	if (buffer == nullptr) ReturnError(CL_INVALID_MEM_OBJECT)
+
+	void *ptr = nullptr;
```

Comment:
The convention is having the "*" paired up with the type, not the identifier.

Reply:
Most of the current code I work with has it the other way around, my muscle memory tends to mess this up.
What do you think about a clang-format file for the repo so code can be auto-formatted in the future (before doing a pull request)?